src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -135,10 +139,16 @@ void Gather::initSupportedPrimitiveDescriptors() {
 
     // Implementation desc type will be redefined in the fn prepareParams if a kernel will be created.
     ov::element::Type dataPrecision = getOriginalInputPrecisionAtPort(GATHER_DATA);
+
+    canOptimizeCompressedEmbedding = false;
+    if ((decompressionSubtractPtr != nullptr) && (decompressionMultiplyPtr != nullptr)) {
+        canOptimizeCompressedEmbedding = true;
+    }
+
     addSupportedPrimDesc({{LayoutType::ncsp, dataPrecision},
```

Comment:
Need to make sure dataPrecision is set correctly for compressed case (need to force U8, otherwise there is no strong gurantee some other transformation will not change it)

Reply:
Add stronger check for dataPrecision == u8